crates/editor/src/editor.rs

```diff
@@ -5475,9 +5475,18 @@ impl Editor {
             }
         }
 
+        let mut common_prefix_len = 0;
+        for (a, b) in old_text.chars().zip(new_text.chars()) {
+            if a == b {
+                common_prefix_len += a.len_utf8();
+            } else {
+                break;
+            }
+        }
+
```

Comment:
This does the same thing, but cleaner (imo!)
```suggestion
        let common_prefix_len = old_text
            .as_bytes()
            .iter()
            .zip(new_text.as_bytes())
            .take_while(|(a, b)| a == b)
            .count();
```

Reply:
code in red is exactly as it was before https://github.com/zed-industries/zed/pull/28586

but more importantly, I think you did .as_bytes() + count(), which should be slightly different then that += with a.len_utf8()

the exact equivalent: 

```suggestion
        let common_prefix_len = old_text
            .chars()
            .iter()
            .zip(new_text.chars())
            .take_while(|(a, b)| a == b)
            .map(|(a, _)| a.len_utf8())
            .sum::<usize>();
```

I think

Reply:
You're right, good catch! I've removed my suggestion to avoid mistakes. I still like your suggestion more than the explicit loop, fwiw!